sql/moz-fx-data-shared-prod/telemetry_derived/firefox_android_clients_v1/query.sql

```diff
@@ -0,0 +1,482 @@
+-- Query first observations for Firefox Android Clients.
+-- Proposal:
+-- https://docs.google.com/document/d/12bj4DhCybelqHVgOVq8KJlzgtbbUw3f68palNrv-gaM/edit#
+CREATE TEMP FUNCTION get_first_not_null_value(
+  existing_value STRING,
+  existing_date DATE,
+  first_session_value STRING,
+  first_session_date DATE,
+  metrics_value STRING,
+  metrics_date DATE
```

Comment:
Could we instead take this is a priority sorted list? Something like:
```
CREATE TEMP FUNCTION get_first_not_null_value(
    date_values ARRAY<STRUCT<value, date>>
)
```
with an unnest and taking the first non-null value for the lowest date. Then for e.g. `source` we'd be able to avoid the `NULL, NULL` arguments.